### PR TITLE
bfcp: Add support for TCP transport for BFCP

### PR DESCRIPTION
**Pull request by Lastique**
This test relies on and tests the updated BFCP APIs:

https://github.com/baresip/re/pull/411

### CODE REVIEW

**Comment by alfredh** [comment]
there is an error running retest, please add -v (verbose) to see what is going on

**Comment by Lastique** [comment]
It's the OOM test failing because nothing gets received on the socket and `re_main` times out. I'm not sure what to do about it. The functional tests pass.

File: src/bfcp.c [hunk 1]
```diff
@@ -232,3 +234,207 @@ int test_bfcp_bin(void)
...
+	TEST_ERR(err);
+	srv->peer = cli->addr;
+
+	err = bfcp_request(cli->bfcp, &cli->peer, BFCP_VER1, BFCP_HELLO, 0u,
+		0u, &response_handler, cli, 0u);
+	TEST_ERR(err);
+
+	err = re_main_timeout(100);
+	TEST_ERR(err);
+
```
Comment (alfredh) [inline src/bfcp.c:375]: if the test could not complete because of resource error e.g. OOM,
you can detect it here.

Comment (Lastique) [reply src/bfcp.c:383]: It doesn't work for UDP since there's no connection timeout error code. I have disabled this test for OOM.

**Pull request merged**